### DISCUSSION TITLE
Why retrive/continuePropertiesEx are not implemented in PropertyCollector class

### DISCUSSION THREAD

**Discussion by KamranMaharov** [vmware/govmomi#3002] [Community Help] [answered]
Dear Govmomi Community,

I have one question regarding which I need your help.
Currently we are trying to migrate old vijava-based codebase to govmomi-based codebase.
I noticed that PropertyCollector class in vijava use retrieve/continuePropertiesEx APIs, but PropertyCollector class in govmomi use retrivePropties API (without ex suffix). In an attempt to try to understand differences I read PropertyCollector section from vSphere documentation, where it says vCenter may return partial data with token if amount of data returned is too much. That's why retrive/continuePropertiesEx is recommended way. 
I'm wondering why you decided to not implement retrive/continuePropertiesEx methods in PropertyCollector class.
Can we rely on RetriveProperties (without ex suffix) method of govmomi.PropertyCollector if data retrieved is too much?
Also noticed that retrieveProperties API is deprecated here:
https://vdc-download.vmware.com/vmwb-repository/dcr-public/fe08899f-1eec-4d8d-b3bc-a6664c168c2c/7fdf97a1-4c0d-4be0-9d43-2ceebbc174d9/doc/vmodl.query.PropertyCollector.html#retrievePropertiesEx
with a little note to use RetrievePropertiesEx instead
```
Deprecated. As of vSphere API 4.1, use [RetrievePropertiesEx](https://vdc-download.vmware.com/vmwb-repository/dcr-public/fe08899f-1eec-4d8d-b3bc-a6664c168c2c/7fdf97a1-4c0d-4be0-9d43-2ceebbc174d9/doc/vmodl.query.PropertyCollector.html#retrievePropertiesEx).
```
Any help is much appreciated.

**Comment by dougm** [comment, marked as answer]
Hi @KamranMaharov , you can use these methods from govmomi, see: https://github.com/vmware/govmomi/issues/2409#issuecomment-833018528
